### PR TITLE
executor: Make executor GCP images public

### LINKED ISSUE #25095: executors: Publish GCP image to the public
Ensure published GCP image can be referenced outside of our sourcegraph-ci project without specific credentials.

### PR DESCRIPTION
**Pull request by efritz**
Fixes #25095.

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
Should it still live in this project then?

**Comment by efritz** [comment]
> Should it still live in this project then?

cc @sourcegraph/distribution @sourcegraph/security We want to publish the executor google compute image as public - which project should an artifact like this live in?

**Comment by andreeleuterio** [comment]
We have most images on the sourcegraph-dev project and some are published to DockerHub as well. I'm hot-potatoing this to Distro because I don't know how we make that call. All good from Security either way.

**Comment by efritz** [comment]
FYI, this is to make the Google VM image (not a docker image) public. We'll need to do the same thing with the AWS AMI generated in packer as well.

**Comment by andreeleuterio** [comment]
Ah crap nvm me then!

File: enterprise/cmd/executor/release.sh [hunk 1]
```diff
@@ -1,4 +1,10 @@
 #!/usr/bin/env bash
 
-# Add released label to the image built by the build.sh command
-gcloud compute images add-labels --project=sourcegraph-ci "executor-$(git log -n1 --pretty=format:%h)-${BUILD_TIMESTAMP}" --labels='released=true'
+# Construct image built by build.sh
```
Comment (bobheadxi) [inline enterprise/cmd/executor/release.sh:3]:
```suggestion
# Point to GCP boot disk image built by build.sh
```

First part: for noobs like me who get caught off guard when an image isn't _gasp_ a docker image

Second part: it looks like this is publish an image that should be built separately by `build.sh` correct?

**Review by bobheadxi — COMMENTED** [review]
I believe this is the right place to have this script for non-private stuff

**Review by daxmc99 — APPROVED** [review]
LGTM, this will still be built manually though, correct? 
Not as part of CI?

**Comment by efritz** [comment]
@daxmc99 This is currently being built in the `main` branch: https://buildkite.com/sourcegraph/sourcegraph/builds/108990#9647c033-5914-462e-9823-ec19c40ee4f6

File: enterprise/cmd/executor/release.sh (hunk 1 above)
Comment (efritz) [reply enterprise/cmd/executor/release.sh:3]: Yes - build.sh submits a job to cloud build to build a GCP boot disk image as well as an AWS AMI containing what is necessary to run the executor service.

**Comment by daxmc99** [comment]
Looks good, I don't know much about this but we typically run tagged versions of this or "latest"?

**Comment by efritz** [comment]
Right now the executors are being deployed by hand as we don't have a good way to set up terraform to run on commit for Dogfood/Cloud environments. We just update it to whatever GCP image/AMI id we need to deploy by hand and re-apply the terraform description.

We'll want to eventually have these images public so that we can point to the correct tagged version built alongside the tagged version of Sourcegraph and distribute those in tandem.

Note that AMIs being public only allow you to use them but not list them from another org. Not sure about GCP, but these are meant to be _runnable_ by customers but nothing else.

**Comment by daxmc99** [comment]
Gotcha, all that makes sense. I will keep an eye out for automating Terraform for the purpose of keeping these images up to date.

**Pull request merged**